warehouse/core/src/main/java/datawave/util/TableName.java

```diff
@@ -0,0 +1,23 @@
+package datawave.util;
+
+/**
+ * Class that contains default table names
+ */
+public interface TableName {
```

Comment:
@tomnelson I disagree with making this an interface. While yes, you can remove the access modified keywords, it really doesn't make any sense. What would it mean to implement TableName?

Reply:
So you think it has to be a (final) class just to prevent some misguided future programmer from extending/implementing it, and you're willing to let it be more cluttered with extraneous declarations just to outwit that future troublemaker?
There's no rule that says an interface must be implemented.

Reply:
Honestly yes I would prefer that. Using an interface for this goes directly against the definition of an interface. 

On top of that, in the future should some developer see this and choose to implement the interface, the values could be silently overridden by members of the implementing class. I understand that this pr does not Implement the class, but the minor amount of clutter caused by having the modifiers is worth protecting from other issues in the future.

Reply:
Good discussion, I'm leaning towards making it a final class and will have an update shortly.

Reply:
i'm fine with leaving it up to the author

Reply:
A table name as a representation of the code has always seem short sighted -- a table name should be a representation of its utility. As a result it should be pulled from configuration or translated dynamically . The virtue of data tables being sharded should be a feature and not part of its nomenclature. Intrinsically you have a data store of information and its indices. As such I think the project would be better suited by not making constants from table names but instead defining a mechanism to translate ( either by config or dynamically ) their names and references. 

The academic argument above, not withstanding, the string location has been condensed, but a re-compilation change would need to be made. One may argue, "that wasn't the original intent," but the same level of effort would be required for this and any other true improvement.

Reply:
Also, it looks clumsy to me now.
Using a class for this goes directly against the definition of a class (to provide a template for the creation of Objects)

Reply:
Sounds like a good improvement, you should write up an issue for it!

Reply:
If it helps the discussion, Josh Bloch says using an interface for constants is an anti-pattern (see https://stackoverflow.com/questions/2659593/what-is-the-use-of-interface-constants).
Seems like an enum could be well suited for use here.

Reply:
I have seen that. My interpretation is that **implementing** an interface for its constants is the anti-pattern, not the mere existence of a structure to hold those constants. I prefer a clean structure with sensible use over a more cluttered class that is really not a 'class'. ymmv.